vyper/codegen/core.py

```diff
@@ -1243,3 +1242,15 @@ def clamp2(lo, arg, hi, signed):
         LE = "sle" if signed else "le"
         ret = ["seq", ["assert", ["and", [GE, arg, lo], [LE, arg, hi]]], arg]
         return IRnode.from_list(b1.resolve(ret), typ=arg.typ)
+
+
+# make sure we don't overrun the source buffer, checking for overflow:
+# valid inputs satisfy:
+#   `assert !(start+length > src_len || start+length < start)`
+def slice_bounds_check(start, length, src_len):
```

Comment:
maybe this needs a new name now that it's not just used for slice. how about something like `check_buffer_overflow_ir()`?

Reply:
ok.
I ended up using the original slice method because I realized that here we're also checking that some buffer slice is within bounds .. as such I kept the name